### ISSUE TITLE
Nesting idb_sys' on_success handlers leads to calling a dropped Closure

### ISSUE THREAD

**Issue by Ekleog** [devashishdxt/idb#17]
https://github.com/devashishdxt/idb/blob/main/idb-sys/src/request/store_request.rs#L40 is the problematic line: it assumes that the `StoreRequest` will stay alive until the callback is called.

However, when nesting on_success handlers (AFAICT the only official way to deal with transactions with multiple requests), the `StoreRequest` will be dropped before the `on_success` handler gets called, leading to a panic.

**Comment by devashishdxt** [comment]
I'm not sure if I understand your concern correctly. Can you write a small test which panics?

**Comment by Ekleog** [comment]
I don't have the infrastrure up to write a small test, and I moved my code back to web-sys after hitting this issue, but here is the pseudo-code that was panicking:
```rust
let db = Database::open("foo");
let transaction = db.transaction();
let get_req = transaction.get("foo").unwrap();
get_req.on_success(|evt| {
    let transaction = evt.do_lots_of_magic_to_get_transaction().unwrap();
    let get_req = transaction.get("bar").unwrap();
    get_req.on_success(|evt| {
        // This will never be called, because the Closure will be dropped before it gets called
        // This is due to `get_req` being dropped at the end of the `on_success` closure, rather than
        // at the end of its real lifetime, which is until the closure actually gets called.
    });
});
```

Hopefully this makes the issue clearer!

In my code I finally decided for a "closure stash", a big Vec that I'll keep alive until my whole transaction has finished executing: https://github.com/Ekleog/crdb/blob/main/src/client/indexed_db.rs#L120

This would, incidentally, probably be how I'd implement https://github.com/devashishdxt/idb/issues/7#issuecomment-1889895131 too :)

**Comment by devashishdxt** [comment]
Fixed in #20.
